metadata-ingestion/src/datahub/cli/ingest_cli.py

```diff
@@ -280,69 +306,96 @@ def deploy(
         resolve_env_vars=False,
     )
 
+    deploy_options_raw = pipeline_config.pop("deployment", None)
+    if deploy_options_raw is not None:
+        deploy_options = DeployOptions.parse_obj(deploy_options_raw)
+
+        logger.info(f"Using {repr(deploy_options)}")
+
+        if urn:
+            raise click.UsageError(
+                "Cannot specify both --urn and deployment field in config"
+            )
+        elif name:
+            raise click.UsageError(
+                "Cannot specify both --name and deployment field in config"
+            )
+        else:
+            logger.info(
+                "The deployment field is set in the recipe, any CLI args will be ignored"
```

Comment:
@hsheth2 IMHO this is odd behavior. Typically command-line arguments override config options, which is probably what most would expect. It would also simplify the logic here - first initialize `deploy_options` from the config and then set any options that were provided on the command-line.

Reply:
yeah I agree - will make some tweaks